src/search_decoder.py

```diff
@@ -42,10 +41,10 @@ def forward(self, input_seq, input_length, max_length):
         all_tokens = torch.zeros([0], device=device, dtype=torch.long)
         all_scores = torch.zeros([0], device=device)
         # Iteratively decode one word token at a time
-        for _ in range(max_length):
+        for _ in range(config.MAX_LENGTH):
```

Comment:
interestingly, the `max_length` is used in `GreedySearch` but not used in `BeamSearch`
not sure if it matters

Reply:
Not sure what is the equivalent thing in `BeamSearch` as `max_length`, might be [this line](https://github.com/aobo-y/SouthPark-Chatbot/pull/22/files#diff-2d4c06dddb2aa04e700a1a6c207449d9R120). We can fix it later.

Reply:
Yes, I think it matters. Maybe we need to change the `While True` condition in: https://github.com/aobo-y/SouthPark-Chatbot/blob/2a2165b9053f0d31ff0fc311c8605e6f019202bb/src/search_decoder.py#L118-L120